SMLHelper/Patchers/BioReactorPatcher.cs

```diff
@@ -2,13 +2,12 @@
 {
     using Harmony;
     using System.Collections.Generic;
-    using Utility;
 
     internal class BioReactorPatcher
     {
         #region Internal Fields
 
-        internal static Dictionary<TechType, float> CustomBioreactorCharges = new Dictionary<TechType, float>(TechTypeExtensions.sTechTypeComparer);
+        internal static IDictionary<TechType, float> CustomBioreactorCharges = new SelfCheckingDictionary<TechType, float>("CustomBioreactorCharges", TechTypeExtensions.sTechTypeComparer);
```

Comment:
`internal static SelfCheckingDictionary<TechType, float>`

Reply:
Nothing outside of the `SelfCheckingDictionary` itself needs to be concerned with what type it is.  
Given its usages, these could likely all be exposed as simple `ICollection`s instead of even the full range of options given to `IDictionary`.

All this is in keeping with **Principle of least privilege**.  
https://en.wikipedia.org/wiki/Principle_of_least_privilege

Reply:
Marking this as resolved. Further cleanup of this kind can and will be done over time.